src/index.js

```diff
@@ -16,5 +16,64 @@ module.exports = {
    * This gives you an opportunity to set up your data model,
    * run jobs, or perform some special logic.
    */
-  bootstrap(/*{ strapi }*/) {},
-};
+  bootstrap({ strapi }) {
+
+    strapi.db.lifecycles.subscribe({
+      models: ["plugin::users-permissions.user"],
+      // afterCreate lifecycle to create a new team-membership entry when a new user is created
+      async afterCreate({result}){
+        const {projects} = result
+
+        for (const project of projects) {
+          await strapi.service("api::team-membership.team-membership").create({
+            data: { 
+              user: [result.id],
+              project: [project.id],
+              joinDate: new Date()
+            }
+          })
+        }
+      },
+      
+      async beforeUpdate({params}) {
+        const joinProject = params.data.projects.connect; 
```

Comment:
What is `projects.connect`?

Reply:
The `beforeUpdate` lifecycle holds the information about the new update that is about to be done on the user collection in the `params.data` object. This object also contains the projects object (`params.data.projects`). In the `parama.data.projects` objects, there are 2 arrays of objects [disconnect ] and [connect]:

`params.data.projects.disconnect` = List of projects removed by the user
`params.data.projects.connect` = List of projects added by the users